crates/ol/mempool/src/handle.rs

```diff
@@ -0,0 +1,465 @@
+//! Mempool service handle for external interaction.
+
+use std::sync::Arc;
+
+use strata_identifiers::{OLBlockCommitment, OLTxId};
+use strata_service::{CommandHandle, ServiceBuilder, ServiceMonitor};
+use strata_status::StatusChannel;
+use strata_storage::NodeStorage;
+use strata_tasks::TaskExecutor;
+use tracing::warn;
+
+use crate::{
+    MempoolCommand, OLMempoolError, OLMempoolResult,
+    command::create_completion,
+    service::{MempoolService, MempoolServiceStatus},
+    state::{MempoolContext, MempoolServiceState},
+    types::{OLMempoolConfig, OLMempoolStats, OLMempoolTransaction},
+};
+
+/// Handle for interacting with the mempool service.
+#[derive(Debug)]
+pub struct MempoolHandle {
+    command_handle: Arc<CommandHandle<MempoolCommand>>,
+    monitor: ServiceMonitor<MempoolServiceStatus>,
+}
+
+impl MempoolHandle {
+    /// Helper to map send/recv errors to ServiceClosed.
+    fn service_closed_error<T>(_: T) -> OLMempoolError {
+        OLMempoolError::ServiceClosed("Mempool service closed".to_string())
+    }
+
+    /// Send command and wait for response.
+    async fn send_command<R>(
+        &self,
+        command: MempoolCommand,
+        rx: tokio::sync::oneshot::Receiver<R>,
+    ) -> OLMempoolResult<R> {
+        self.command_handle
+            .send(command)
+            .await
+            .map_err(Self::service_closed_error)?;
+
+        rx.await.map_err(Self::service_closed_error)
+    }
+
+    /// Create and launch a new mempool service.
+    ///
+    /// # Arguments
+    /// * `storage` - Node storage (provides both database manager and StateAccessor creation)
+    /// * `config` - Mempool configuration
+    /// * `current_tip` - Current chain tip (slot + block ID)
+    /// * `status_channel` - Status channel for subscribing to chain sync updates
+    /// * `texec` - Task executor for spawning the service task
+    ///
+    /// # Returns
+    /// A handle to interact with the launched service
+    pub async fn launch(
+        storage: Arc<NodeStorage>,
+        config: OLMempoolConfig,
+        current_tip: OLBlockCommitment,
+        status_channel: StatusChannel,
+        texec: &TaskExecutor,
+    ) -> anyhow::Result<Self> {
```

Comment:
This should be split out into its own function, maybe in a builder module.  This being on the `MempoolHandle` type implies that the thing we're focusing on is the handle itself, not the mempool service that the handle is controlling.

Reply:
Similar to the `AsmWorkerHandle`?